File: TheCProgrammingLanguage/src/chapter_06/point.c
/**
 * The C Programming Language
 * 
 * version: 2nd
 * author: Brian W.Kernighan
 *         Dennis M.Ritchie
 */

#include <stdio.h>
#include <math.h>

struct point {
    int x;
    int y;
};

int main() {
    struct point pt = {3, 4};
    double dist;

    dist = sqrt((double)pt.x * pt.x + (double)pt.y * pt.y);

    printf("%d, %d\n", pt.x, pt.y);
    printf("the distance is %.2f", dist);

    return 0;
}

/*
output:
3, 4
the distance is 5.00
*/

File: JavaTutorial/src/chapter_05/Point.java
/**
 * Java Tutorial
 * 
 * @author
 * @version 1st
 * @see 
 */

/**
 * 测试类
 */
public class Point {
    // 类属性
    private double x;
    private double y;

    // 构造方法
    public Point(double x, double y) {
        this.x = x;
        this.y = y;
    }

    // 类方法

    /**
     * 设置新坐标
     * @param x 横坐标
     */
    public void setX(double x) {
        this.x = x;
    }
    /**
     * 设置新坐标
     * @param y 纵坐标
     */
    public void setY(double y) {
        this.y = y;
    }

    public double getDistance(Point p) {
        return Math.sqrt((x-p.x)*(x-p.x) + (y-p.y)*(y-p.y));
    }

    /**
     * main()
     * 
     * @param args array of string arguments
     */
    public static void main(String[] args) {
        // 创建对象
        Point p1 = new Point(3.0, 4.0);
        Point p2 = new Point(0.0, 0.0);
        System.out.println(p1.getDistance(p2));
    }
}

/**
 * output:
 * 
 */